enhancements/two-nodes-openshift/2no.md

```diff
@@ -0,0 +1,486 @@
+Should the admin wish for the control-plane to start, the admin will need to execute a supplied confirmation command on the active cluster node.
+This command will grant quorum to the RHEL-HA components, authorizing it to fence its peer and start etcd as a cluster-of-one in read/write mode.
+Confirmation can be given at any point and optionally make use of SSH to facilitate initiation by an external script.
+
+### API Extensions
+
+There are two related but ultimately orthogonal capabilities that may require API extensions.
+
+1. Identify the cluster as having a unique topology
+2. Tell CEO when it is safe for it to disable certain membership-related functionalities
+
+#### Unique Topology
+
+A mechanism is needed for components of the cluster to understand that this is a 2-node control-plane topology which may require different handling.
+We will define a new value for the `TopologyMode` enum: `DualReplica`.
+The enum is used for the `controlPlaneTopology` and `infrastructureTopology` fields, and the currently supported values are `HighlyAvailable`, `SingleReplica`, and `External`.
+
+We will additionally define a new feature gate `DualReplicaTopology` that can be enabled in `install-config.yaml` to ensure the feature can be set as `TechPreviewNoUpgrade`.
+
+#### CEO Trigger
+
+Initially, the creation of an etcd cluster will be driven in the same way as other platforms.
+Once the cluster has two members, the etcd daemon will be removed from the static pod definition and recreated as a resource controlled by RHEL-HA.
+At this point, the Cluster Etcd Operator (CEO) will be made aware of this change so that some membership management functionality that is now handled by RHEL-HA can be disabled.
+This will be achieved by having the same entity that drives the configuration of RHEL-HA use the OpenShift API to update a field in the CEO's `ConfigMap` - which can only succeed if the control-plane is healthy.
+
+To enable this flow, we propose the addition of a `managedEtcdKind` field which defaults to `Cluster` but will be set to `External` during installation, and will only be respected if the `Infrastructure` CR's `TopologyMode` is `DualReplicaTopologyMode`.
+This will allow the use of a credential scoped to `ConfigMap`s in the `openshift-etcd-operator` namespace, to make the change.
+
+### Topology Considerations
+
+2NO represents a new topology and is not appropriate for use with HyperShift, SNO, or MicroShift
+
+#### Standalone Clusters
+
+Two-node OpenShift is first and foremost a topology of OpenShift, so it should be able to run without any assumptions of a cluster manager.
+
+### Implementation Details/Notes/Constraints
+
+While the target installation requires exactly 2 nodes, this will be achieved by proving out the "bootstrap plus 2 nodes" flow in the core installer and then using the Assisted Installer's ability to bootstrap from one of the target machines to remove the requirement for a bootstrap node.
```

Comment:
Note that this might require an additional bootstrap strategy in CEO (as was required for assisted ability to install 3 nodes while reusing the bootstrap node as the 3rd control plane).
essentially CEO needs to know how many members to expect in order to say that it's ready (the installation script wait for it [here](https://github.com/openshift/installer/blob/0d294cffef478e7d9f8a8484cad69cc34896cad3/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L523))
@tjungblu ?

Reply:
it does, @jaypoulz has it on its list as far as I understood